### PR TITLE
storage/etcd: no-op, simplify watcher construction

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Simplifies the construction of a watcher by eliminating the need for a separate constructor function and enabling direct instantiation of the watcher. It will be used in https://github.com/kubernetes/kubernetes/pull/119557

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
https://github.com/kubernetes/enhancements/issues/3157

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go [hunk 1]
```diff
@@ -87,34 +86,18 @@ type watchChan struct {
 	errChan           chan error
 }
 
-func newWatcher(client *clientv3.Client, codec runtime.Codec, groupResource schema.GroupResource, newFunc func() runtime.Object, versioner storage.Versioner) *watcher {
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go:90]: I'm planing to add more fields to the `watcher` struct in https://github.com/kubernetes/kubernetes/pull/119557.

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go [hunk 2]
```diff
@@ -122,10 +122,25 @@ func newStore(c *clientv3.Client, codec runtime.Codec, newFunc func() runtime.Ob
...
+		versioner:     versioner,
+		transformer:   transformer,
+	}
+	if newFunc == nil {
+		w.objectType = "<unknown>"
+	} else {
+		w.objectType = reflect.TypeOf(newFunc()).String()
+	}
+
+	s.watcher = w
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go:142]: Why not create watcher first (it doesn't depend on s) and then just set is when creating store (as it was before)?

Comment (p0lyn0mial) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go]: np, pushed, PTAL.

**Pull request merged**